.github/workflows/lint-and-test.yml

```
@@ -193,7 +193,7 @@ jobs:
       - name: Run tests with pytest
         run: pytest --cov-report=xml
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
```

```suggestion
        uses: codecov/codecov-action@v4
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
```

I’ve added this as a repository secret for you.